src/form_components/repeating_group.coffee

```diff
@@ -0,0 +1,155 @@
+        @set('skipped', true)
+
+    @entries = _.map entryValues, (value) =>
+      new FormRenderer.Models.ResponseFieldRepeatingGroupEntry({ value }, @fr, @)
+
+  addEntry: ->
+    @entries.push(
+      new FormRenderer.Models.ResponseFieldRepeatingGroupEntry({}, @fr, @)
+    )
+
+  removeEntry: (idx) ->
+    @entries.splice(idx, 1)
+
+    if @entries.length == 0
+      @set('skipped', true)
+
+  isSkipped: ->
+    !!@get('skipped')
+
+  getValue: ->
+    if @isSkipped()
+      []
+    else
+      _.invoke @entries, 'getValue'
+
+  maxEntries: ->
+    if @get('maxentries')
+      parseInt(@get('maxentries'), 10) || Infinity
+    else
+      Infinity
+
+  canAdd: ->
+    @entries.length < @maxEntries()
+
+FormRenderer.Models.ResponseFieldRepeatingGroupEntry = Backbone.Model.extend
+  initialize: (_attrs, @fr, @repeatingGroup) ->
+    @initFormComponents @repeatingGroup.get('children'), @get('value') || {}
+
+  reflectConditions: ->
+    @view.reflectConditions()
+
+  canRemove: ->
+    @repeatingGroup.entries.length > 1
+
+FormRenderer.Views.ResponseFieldRepeatingGroup = Backbone.View.extend
+  className: 'fr_response_field fr_response_field_group'
+
+  events:
+    'click .js-remove-entry': 'removeEntry'
+    'click .js-add-entry': 'addEntry'
+    'click .js-skip': 'toggleSkip'
+
+  initialize: (options) ->
+    @_sharedInitialize(options)
+
+    # Forward `shown` and `hidden` events to subviews
+    @on 'shown', => view.trigger('shown') for view in @views
+    @on 'hidden', => view.trigger('hidden') for view in @views
+
+  toggleSkip: ->
+    @model.set('skipped', !@model.isSkipped())
+
+    # When clicking "answer this question", add an entry if there are none
+    if !@model.isSkipped() && @model.entries.length == 0
+      @addEntry()
+
+    @render()
+
+  addEntry: ->
+    @model.addEntry()
+    @render()
+    _.last(@views).focus()
+
+  removeEntry: (e) ->
+    idx = @$el.find('.js-remove-entry').index(e.target)
+    @model.removeEntry(idx)
+    @render()
+
+  render: ->
+    @views = []
+    $els = $()
+
+    for entry, idx in @model.entries
+      view = new FormRenderer.Views.ResponseFieldRepeatingGroupEntry(
+        entry: entry,
+        form_renderer: @form_renderer,
+        idx: idx
+      )
+
+      entry.view = view
+      $els = $els.add view.render().el
+      @views.push view
+
+    @$el.html JST['partials/repeating_group'](@)
+    rivets.bind @$el, { @model }
+    @$el.find('.fr_group_entries').append($els)
+    @form_renderer?.trigger 'viewRendered', @
+    @
+
+FormRenderer.Views.ResponseFieldRepeatingGroupEntry = Backbone.View.extend
+  className: 'fr_group_entry'
+
+  initialize: (options) ->
+    @entry = options.entry
+    @form_renderer = options.form_renderer
+    @idx = options.idx
+    @views = []
+
+    # Forward `shown` and `hidden` events to subviews
+    @on 'shown', => view.trigger('shown') for view in @views
+    @on 'hidden', => view.trigger('hidden') for view in @views
```

Comment:
what are the subviews of a repeating group entry?

Reply:
I don't think repeating group entries ever have subviews, but I left this in place because it seemed to be for future-proofing in case they did. 

Right now this doesn't do anything, so I'd be happy to remove it.

Reply:
i don't have a strong preference either way